shell/platform/darwin/ios/framework/Source/FlutterEngineTest.mm

```diff
@@ -6,12 +6,17 @@
 #import <XCTest/XCTest.h>
 #include "flutter/shell/platform/darwin/common/framework/Headers/FlutterMacros.h"
 #import "flutter/shell/platform/darwin/ios/framework/Headers/FlutterEngine.h"
+#import "flutter/shell/platform/darwin/ios/framework/Source/FlutterBinaryMessengerRelay.h"
 
 FLUTTER_ASSERT_ARC
 
 @interface FlutterEngineTest : XCTestCase
 @end
 
+@interface FlutterEngine (Test) <FlutterBinaryMessenger>
```

Comment:
Put this category in a headers so tests don't have to redeclare this.

Reply:
what implementation style were you thinking? Having something like `Source/FlutterEngine.mm`, `Source/FlutterEngine_Internal.h`, `Source/FlutterEngine_TestFlutterBinaryMessenger.h`, `Source/FlutterEngine_TestSomeOtherProtocol.h`, `Source/FlutterEngineTest.mm`? 

We might end up with a big list since no other test is using this specific category yet (Y(probably)AGNI). 

We can also put all "internals" into one header, but it's not very least-privilege-principle.

Reply:
I think FlutterEngine_Test sounds fine.  It would operate like java's visiblefortesting.  Everything can just go in there.

I'd say FlutterEngine_Test is like visiblefortesting and FlutterEngine_internal is like package visible methods.